bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java

```diff
@@ -95,7 +97,18 @@ public RepositoryTransport(IProvisioningAgent agent) {
 	@Override
 	public IStatus download(URI toDownload, OutputStream target, IProgressMonitor monitor) {
 		boolean promptUser = false;
+
 		boolean useJREHttp = false;
+		try {
+			// DELMARVA - is org.eclipse.core.pki installed and configured
+			if (SSLContext.getDefault().getProvider().getName().contains("TLS")) { //$NON-NLS-1$
+				useJREHttp = true;
+				Activator.getDefault().useJREHttpClient();
+			}
+		} catch (NoSuchAlgorithmException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
Please use `ILog.log().error(e.getMessage(), e)`

Reply:
@mickaelistria  I noticed that most of the other try catch blocks just throw and dont print anything out.  There are no other ILog messages in that bundle.  Maybe better to stick with the picture show and just throw?

Reply:
@mickaelistria   Looking again, the software is supposed to be transparent and not required, so perhaps btter to not do anything in the catch block.  Software does not even need to know an exception may have occurred.

Reply:
Please 🙏 try to keep a PR to a single commit. Amend and force push further changes. Then you won’t be asked to squash the commits later.